### PR TITLE
Fix/lp 195 refactor components

### PR DESCRIPTION
**Pull request by irv205**

### CODE REVIEW

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 1]
```diff
@@ -32,19 +32,14 @@ import com.iteneum.designsystem.components.phonenumbertext.PhoneNumberTransforma
 @Composable
 fun LpOutlinedTextFieldPassword(
     modifier: Modifier,
-    value: String,
+    value: String = "",
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:38]: Please check in which other palces is using the value

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 2]
```diff
@@ -98,16 +93,13 @@ fun LpOutlinedTextField(
     hint: String,
     isValid: Boolean,
     supportTextError: String,
+    value: String = "",
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:102]: Please update the documentation

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 3]
```diff
@@ -164,16 +156,13 @@ fun LpOutlinedTextFieldMail(
     label: String,
     isValid: Boolean,
     supportTextError: String,
+    value: String = "",
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:167]: Please update the documentation

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 4]
```diff
@@ -222,15 +211,12 @@ fun LpOutlinedTextFieldInput(
     enabled: Boolean = true,
     label: String,
     hint: String,
+    value: String = "",
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:224]: Please update the documentation

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 5]
```diff
@@ -277,10 +263,12 @@ fun DropdownTextField(
     modifier: Modifier = Modifier,
     title: String,
     items: List<String>,
-    selected: (String) -> Unit
+    selected: (String) -> Unit,
+    value: String = "",
+    onSelectedOption: (String) -> Unit,
+    newSelectedOption: (String) -> Unit
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: Please update the documentation

Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: Do you really need two high order functions? Can I know why did you created each one please?

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 6]
```diff
@@ -277,10 +263,12 @@ fun DropdownTextField(
     modifier: Modifier = Modifier,
     title: String,
     items: List<String>,
-    selected: (String) -> Unit
+    selected: (String) -> Unit,
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: I think you just need one value or selected and one onSelectedOption.

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 1 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:38]: update, let me know if this works

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 2 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:102]: done

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 3 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:167]: done

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 4 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:224]: done

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 5 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: done

Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: update, let me know if this is a best way

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 6 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: update

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 7]
```diff
@@ -267,20 +261,22 @@ fun LpOutlinedTextFieldInput(
...
  */
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun DropdownTextField(
     modifier: Modifier = Modifier,
     title: String,
     items: List<String>,
-    selected: (String) -> Unit
+    selected: (String) -> Unit,
+    value: String = "",
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:278]: we can remove this comma please

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 8]
```diff
@@ -345,21 +340,22 @@ fun DropdownTextField(
...
 @Composable
 fun LPPhoneNumberText(
     modifier: Modifier = Modifier,
-    value: String,
+    value: String = "",
     isEnabled: Boolean = true,
     isNotValid: Boolean,
     supportTextError: String,
-    onPhoneChanged: (String) -> Unit
+    onPhoneChanged: (String) -> Unit,
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: we can remove the comma please

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt [hunk 9]
```diff
@@ -84,7 +84,7 @@ fun RepairView() {
                 ),
                 title = stringResource(id = R.string.label_pet_in_unit),
                 items = optionsPetInUnit.toList(),
-                selected = {/* TODO - PetInUnit field - To verify if extra functionality required */ }
+                selected = {/* TODO - PetInUnit field - To verify if extra functionality required */ },
```
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we can remove the comma please

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt [hunk 10]
```diff
@@ -99,7 +99,7 @@ fun RepairView() {
                 ),
                 title = stringResource(id = R.string.label_category),
                 items = optionsCategory.toList(),
-                selected = {/* TODO - Category field - To verify if extra functionality required */ }
+                selected = {/* TODO - Category field - To verify if extra functionality required */ },
```
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we can remove the comma please

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 7 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:278]: done

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt (hunk 10 above)
Comment (irv205) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: done

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt (hunk 9 above)
Comment (irv205) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Done

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 8 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: Done

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 11]
```diff
@@ -66,7 +62,7 @@ fun LpOutlinedTextFieldPassword(
             if (isValid) {
                 Text(text = supportTextError)
             } else {
-                Text(text = "")
+                Text(text = value)
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: Here we don't need the value

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 12]
```diff
@@ -27,24 +27,20 @@ import com.iteneum.designsystem.components.phonenumbertext.PhoneNumberTransforma
...
-        value = passwordFieldValue,
-        onValueChange = {
-            passwordFieldValue = it
-            onPasswordChanged(passwordFieldValue.text)
-        },
+        value = value,
+        onValueChange = onPasswordChanged,
         modifier = modifier,
         label = {
             Text(
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:48]: check this text please, what are you writing in the text is going to be show in the name of the text field

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 13]
```diff
@@ -267,20 +261,22 @@ fun LpOutlinedTextFieldInput(
...
  * @param selected refers to a high order function that returns the selected option of the DropDownList
+ * @param value The current value of the input field
  *
  * @author Jesus Lopez Gonzalez
  * @modifiedBy Jose Miguel Garcia Reyes
+ * @modifiedBy Irving Gonzalez
  */
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun DropdownTextField(
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:274]: Please test if it is working

**Review by carlosph816 — APPROVED** [review]

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 11 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: done

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 13 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:274]: This component  works:

image

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 12 above)
Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:48]: Change this part and and some extra parameters **LabelText** and modify **value** parameter:

image

also implement this component in this example:

image

All works good:

image

image

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 14]
```diff
@@ -29,28 +27,25 @@ import com.iteneum.designsystem.components.phonenumbertext.PhoneNumberTransforma
...
  */
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun LpOutlinedTextFieldPassword(
-    modifier: Modifier,
-    value: String,
-    isValid: Boolean,
-    supportTextError: String,
+    modifier: Modifier = Modifier,
+    LabelText: String = "",
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: the name of the parameters should start in lowercase and should be a required parameter so please delete the empty string

Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: Also add the variable to the documentation

Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: done

Comment (irv205) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: done

**Pull request merged**